src/gui/LayoutMapper.cpp

```diff
@@ -19,52 +12,44 @@ LayoutMapper::~LayoutMapper()
 }
 
 
-void LayoutMapper::configureFromSettings(int numPages, Settings* settings)
+void LayoutMapper::configureFromSettings(size_t numPages, Settings* settings)
 {
 	XOJ_CHECK_TYPE(LayoutMapper);
 
+	auto isPairedPages = settings->isShowPairedPages();
+	auto pairsOffset = isPairedPages ? settings->getPairsOffset() : 0;
+
+
+	auto numCols = settings->isPresentationMode() ? 1 : settings->getViewColumns();
+	auto numRows = settings->isPresentationMode() ? 1 : settings->getViewRows();
+	auto fixRows = settings->isPresentationMode() ? false : settings->isViewFixedRows(); //i.e. use columns
+
 	// get from user settings:
-	bool isPairedPages = settings->isShowPairedPages();
-	int numCols = settings->getViewColumns();
-	int numRows = settings->getViewRows();
-	bool fixRows = settings->isViewFixedRows();
-	int pairsOffset = settings->getPairsOffset();
-	bool isVertical = settings->getViewLayoutVert();
-	bool isRightToLeft = settings->getViewLayoutR2L();
-	bool isBottomToTop = settings->getViewLayoutB2T();
-
-	if (!isPairedPages)
+	auto isVertical = settings->getViewLayoutVert();
+	auto isRightToLeft = settings->getViewLayoutR2L();
+	auto isBottomToTop = settings->getViewLayoutB2T();
+
+	if (settings->isPresentationMode())
 	{
-		pairsOffset = 0;
-	}
 
+	};
 	//assemble bitflags for LayoutType
-	int type = isVertical ? LayoutBitFlags::Vertically : 0;
-	type |= isRightToLeft ? LayoutBitFlags::RightToLeft : 0;
-	type |= isBottomToTop ? LayoutBitFlags::BottomToTop : 0;
+	auto type = settings->isPresentationMode() ? Vertical :
+	            isVertical ? Vertical : 0u |
+	                                    isRightToLeft ? RightToLeft : 0u |
+	                                                                  isBottomToTop ? BottomToTop : 0u;
```

Comment:
This is not really nice as it takes some effort to understand. The previous way was better.

Reply:
right its also wrong there are braces missing, i fixed it. It should be way better now. Thanks for the review.